performant-software/neatline-spa
# Editor interface for records

### What does this PR do?
- Adds a tabbed interface for record selecting, editing, creation, and deletion to the editor sidebar in the exhibit show view

### What issues does it address?
- Addresses #6  

### How to test
- After logging in to https://neatline-3-staging.herokuapp.com/, go to https://neatline-3-staging.herokuapp.com/s/public-site/neatline/full to load the full-window base Neatline view
- Create a new exhibit or open an existing one by clicking on its title
- In the exhibit editor, you should see a tabbed interface with exhibit settings in the first tab panel and a records list in the second. Clicking the Records tab and then New Record should open a third tab where you can enter fields for a new record.
- Click save in the new record tab after entering a title to create a new record. The name of the tab should update to the title you entered, and the end of the window URL should update from /edit/new to /edit/{record-id}
- Your new record should appear in the list under the Records tab
- Clicking other records or their shapes on the map (open the "My Demo Exhibit" exhibit for records that actually have map markers) should load those records into the record editor tab. You should be able to close this tab at any point by clicking the 'x' next to its title.
- The window URL should reflect the state of the record editor tab: ending with just /show/{exhibit-slug} when the tab is closed, ending with /edit/new when you've clicked New Record but haven't saved it yet, and ending with /edit/{record-id} when editing an existing record. Refreshing the page at any point should return you to the same tab / record selection state.
- Try editing both the text and style fields for a record, as well as deleting a record.
